[WPE] Auxiliary processes fail to launch when ProcessProviderLibWPE is enabled

zhani opened the pull request:
#### c5dba56a0c36f80749d048c1c24fc67109febd52
<pre>
[WPE] Auxiliary processes fail to launch when ProcessProviderLibWPE is enabled
<a href="https://bugs.webkit.org/show_bug.cgi?id=280321">https://bugs.webkit.org/show_bug.cgi?id=280321</a>

Reviewed by Michael Catanzaro.

The ProcessProviderLibWPE is only used in the UIProcess, but
AuxiliaryProcessMainCommon::parseCommandLine attempts to check if
ProcessProviderLibWPE is enabled. Since ProcessProviderLibWPE is never
enabled for auxiliary processes, this causes a mismatch in the argument count
for auxiliary processes in runtimes where ProcessProviderLibWPE is used.
As a result, the auxiliary process never starts.

* Source/WebKit/Shared/unix/AuxiliaryProcessMain.cpp:
(WebKit::AuxiliaryProcessMainCommon::parseCommandLine):

Canonical link: <a href="https://commits.webkit.org/284344@main">https://commits.webkit.org/284344@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/23ad15f0cbd87b15652eeab6bfd40674ce593512

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/68784 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/48176 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/21443 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/72854 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/19929 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/55972 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/19745 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/54824 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/13259 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/71851 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/44017 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/59393 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/35291 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/40682 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/18287 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/62632 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/17166 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/74548 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/12755 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/16416 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/62311 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/12795 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/59474 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/62348 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/10306 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/3918 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/10556 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/43977 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/45051 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/46245 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/44793 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/WebKit/Shared/unix/AuxiliaryProcessMain.cpp:
```diff
@@ -82,24 +70,30 @@ bool AuxiliaryProcessMainCommon::parseCommandLine(int argc, char** argv)
     return true;
 }
 
+
```
```suggestion
```

zhani replied:
fixed

mcatanzaro commented on Source/WebKit/Shared/unix/AuxiliaryProcessMain.cpp:
```diff
@@ -82,24 +70,30 @@ bool AuxiliaryProcessMainCommon::parseCommandLine(int argc, char** argv)
 #if ENABLE(DEVELOPER_MODE)
-    if (argc > minimumNumArgs && argv[minimumNumArgs] && !strcmp(argv[minimumNumArgs], "--configure-jsc-for-testing"))
-        JSC::Config::configureForTesting();
+    // Check last remainnging options for JSC testing
```
```suggestion
    // Check last remaining options for JSC testing
```

zhani replied:
fixed

mcatanzaro reviewed (approved):
Sorry for breaking this, and thanks for fixing it.

zhani committed:
[WPE] Auxiliary processes fail to launch when ProcessProviderLibWPE is enabled
https://bugs.webkit.org/show_bug.cgi?id=280321

Reviewed by Michael Catanzaro.

The ProcessProviderLibWPE is only used in the UIProcess, but
AuxiliaryProcessMainCommon::parseCommandLine attempts to check if
ProcessProviderLibWPE is enabled. Since ProcessProviderLibWPE is never
enabled for auxiliary processes, this causes a mismatch in the argument count
for auxiliary processes in runtimes where ProcessProviderLibWPE is used.
As a result, the auxiliary process never starts.

* Source/WebKit/Shared/unix/AuxiliaryProcessMain.cpp:
(WebKit::AuxiliaryProcessMainCommon::parseCommandLine):

Canonical link: https://commits.webkit.org/284344@main
